osquery/tables/system/darwin/efigy.cpp

```diff
@@ -0,0 +1,496 @@
+  version.clear();
+
+  auto platform_info = osquery::SQL::selectAllFrom("platform_info");
+  if (platform_info.empty()) {
+    return osquery::Status(1, "Failed to list the platform information");
+  }
+
+  std::stringstream stream(platform_info[0]["version"]);
+
+  std::vector<std::string> fields;
+  for (std::string s; std::getline(stream, s, '.'); fields.push_back(s))
+    ;
+
+  if (fields.size() != 5U) {
+    return osquery::Status(1, "Invalid platform version field");
+  }
+
+  version = fields[0] + "." + fields[2] + "." + fields[3];
+  return osquery::Status(0, "OK");
+}
+
+osquery::Status getSMCVersion(std::string& version) {
+  version.clear();
+
+  auto smc_keys =
+      osquery::SQL::selectAllFrom("smc_keys", "key", osquery::EQUALS, "RVBF");
+  if (smc_keys.empty()) {
+    return osquery::Status(1, "Failed to select the RVBF smc_keys row");
+  }
+
+  std::string raw_version_field = smc_keys[0]["value"];
+  auto part0 = getSMCVersionField(raw_version_field, 0);
+  auto part1 = getSMCVersionField(raw_version_field, 1);
+  auto part2 = getSMCVersionField(raw_version_field, 2);
+  auto part3 = getSMCVersionField(raw_version_field, 3);
+
+  version = part0 + "." + part1 + part2 + part3;
+  if (version.empty()) {
+    return osquery::Status(1, "Failed to retrieve the smc version");
+  }
+
+  return osquery::Status(0, "OK");
+}
+
+osquery::Status getHardwareModel(std::string& model) {
+  model.clear();
+
+  auto mac_system_info = osquery::SQL::selectAllFrom("system_info");
+  if (mac_system_info.empty()) {
+    return osquery::Status(1, "Failed to list the system information");
+  }
+
+  model = mac_system_info[0]["hardware_model"];
+  boost::trim(model);
+  if (model.empty()) {
+    return osquery::Status(1, "Failed to retrieve the hardware model");
+  }
+
+  return osquery::Status(0, "OK");
+}
+
+osquery::Status getOSVersion(std::string& version, std::string& build) {
+  version.clear();
+  build.clear();
+
+  auto sw_vers = osquery::SQL::selectAllFrom(
+      "plist",
+      "path",
+      osquery::EQUALS,
+      "/System/Library/CoreServices/SystemVersion.plist");
+  if (sw_vers.empty()) {
+    return osquery::Status(1, "Failed to parse the SystemVersion plist file");
+  }
+
+  for (const auto& row : sw_vers) {
+    if (row.at("key") == "ProductBuildVersion") {
+      build = row.at("value");
+
+    } else if (row.at("key") == "ProductVersion") {
+      version = row.at("value");
+    }
+  }
+
+  if (version.empty() || build.empty()) {
+    return osquery::Status(
+        1, "Failed to retrieve the OS version and build number");
+  }
+
+  return osquery::Status(0, "OK");
+}
+
+struct IORegistryEntryDeleter final {
+  using pointer = io_registry_entry_t;
+
+  void operator()(pointer p) {
+    IOObjectRelease(p);
+  }
+};
+
+struct CFStringRefDeleter final {
+  using pointer = CFStringRef;
+
+  void operator()(pointer p) {
+    CFRelease(p);
+  }
+};
+
+osquery::Status getBoardID(std::string& board_id) {
```

Comment:
`board_id` would be great to add to the existing `platform_info` table or in a separate table.